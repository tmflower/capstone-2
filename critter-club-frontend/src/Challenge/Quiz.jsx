import { useEffect, useState } from 'react';
import { DietForm } from './DietForm';
import { LocationsForm } from './LocationsForm';
import { TaxClassForm } from './TaxClassForm';
import { ScientificNameForm } from './ScientificNameForm';
import { PreyForm } from './PreyForm';
import usersAPI from '../api/usersAPI';
import { useContext } from 'react';
import UserContext from '../userContext';
import { useNavigate, Link } from 'react-router-dom';

export function Quiz({ 
                    taxClass,
                    setTaxClass, 
                    commonName, 
                    scientificName, 
                    locations, 
                    diet, 
                    phylum, 
                    prey, 
                    setAnimalSelected }) {
    
    const currentUser = useContext(UserContext); 
    const username = currentUser.user.username;
    const userId = currentUser.user.id;
    const navigate = useNavigate();

    const message = {
        correct: "⭐️⭐️⭐️ You got it! ⭐️⭐️⭐️",
        incorrect: "😧😧😧 Sorry, try again. 😧😧😧"
    };

    const [points, setPoints] = useState(0);
    const [numQuestions, setNumQuestions] = useState(0);
    const [animalId, setAnimalId] = useState();
    const [animals, setAnimals] = useState([]);

    useEffect(() => {
        async function getAnimals() {
            const res = await usersAPI.getAllAnimals();
            console.log(res);
            setAnimals(res.animals);
        }
        getAnimals();
    }, [commonName]);
console.log(animals);

    useEffect(() => {
        async function getAnimalId() {
            if(animals.length) { console.log(commonName)
                const animal = animals.filter(animal => commonName === animal.common_name);
                console.log(animal)
                setAnimalId(animal[0].id);
            }
        } 
        getAnimalId();
    },[animals, commonName])


console.log(animalId)

    async function handleSubmit() {
        if (points / numQuestions === 10) {
            alert(`Congratulations, ${username}! You earned ${points} points and the ${commonName} badge!`);
            await usersAPI.updatePoints({ username, points });
            await usersAPI.addBadge({ animalId, userId });
            navigate("/dashboard", {replace: true});
            refreshPage();
        } 
        else {
            alert("Sorry, you lose!");
            navigate("/dashboard", {replace: true});
        }
    }

    function refreshPage(){ 
        window.location.reload(); 
      }
    const handleClick = () => {
        setAnimalSelected(false);
    }

    return (
        <div>
            <h1>Take the {commonName.toUpperCase()} challenge!</h1>
            <DietForm 
                commonName={commonName} 
                diet={diet}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </DietForm>
            <LocationsForm 
                commonName={commonName} 
                locations={locations}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </LocationsForm>
            {phylum === "Chordata" ?  
            <TaxClassForm 
                commonName={commonName} 
                taxClass={taxClass} 
                setTaxClass={setTaxClass} 
                phylum={phylum}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </TaxClassForm> 
            : null}            
            { prey ? 
            <PreyForm 
                commonName={commonName} 
                prey={prey}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </PreyForm> 
            : null }
            {scientificName ?
            <ScientificNameForm 
                commonName={commonName} 
                scientificName={scientificName}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </ScientificNameForm>  
            : null }   
            <button type="button" onClick={handleClick}>Return to animal</button>
            { numQuestions >= 3 ?           
            <button type="submit" onClick={handleSubmit}>Submit Your Answers</button> : null }
            <Link to="/dashboard"></Link>
        </div>
    )
}