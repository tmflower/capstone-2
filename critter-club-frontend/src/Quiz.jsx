import { useState, useEffect } from 'react';
import AnimalsAPI from './api/animalsAPI';


export function Quiz() {

    const [animal, setAnimal] = useState(null);
    const [commonName, setCommonName] = useState("");
    // habitat data is too messy for quiz; remove from here but leave in animal info/card
    // const [habitat, setHabitat] = useState("");
    const [phylum, setPhylum] = useState("");
    const [taxClass, setTaxClass] = useState("");
    const [locations, setLocations] = useState([]);
    const [diet, setDiet] = useState("");
    // const [photo, setPhoto] = useState("");
    
    // Where will points come into play?
    const animalName = localStorage.getItem("animal");

    useEffect(() => {
        async function getAnimal() {
            setAnimal(await AnimalsAPI.getSingleAnimal(animalName))
        }
        getAnimal();
    }, [animalName])

    useEffect(() => {
        if (animal) {       
            setCommonName(animal.name);
            // setGroup(animal.characteristics.group);
            setPhylum(animal.taxonomy.phylum);
            // setHabitat(animal.characteristics.habitat);
            setLocations(animal.locations.map(location => location.replace(/[^a-z]/gi, "")));
            setDiet(animal.characteristics.diet);
            setTaxClass(animal.taxonomy.class);
        }
    }, [animal])

    // console.log("class", taxClass)

    if (taxClass === "Sarcopterygii" || taxClass === "Actinopterygii" || taxClass === "Chondrichthyes") {
        setTaxClass("Fish");
    }

    const validLocations = [ 
        "Africa",
        "Antarctica",
        "Asia",
        "Europe",
        "NorthAmerica",
        "Eurasia",
        "Ocean",
        "CentralAmerica",
        "SouthAmerica",
        "Oceania"
   ];

    const initialState = { 
        vertebrateGroup: '', 
        animalDiet: '',
        Africa: false,
        Antarctica: false,
        Asia: false,
        Europe: false,
        NorthAmerica: false,
        Eurasia: false,
        Ocean: false,
        CentralAmerica: false,
        SouthAmerica: false,
        Oceania: false
    }

    const [formData, setFormData] = useState(initialState);
    const { 
        vertebrateGroup, 
        animalDiet, 
        Africa, 
        Antarctica, 
        Asia,
        Europe,
        NorthAmerica,
        Eurasia,
        Ocean,
        CentralAmerica,
        SouthAmerica,
        Oceania
    } = formData;

    const handleChange = (e) => {
        const { name, value, type, checked } = e.target;
		setFormData((formData) => 
            ({ ...formData, [name]: type === "checkbox" ? checked : value }));
    }

    // problem with checking answers when one question set is excluded. Can submit each separately or figure out how to evaluate conditionally
    const handleSubmit = (e) => {
        e.preventDefault();
        let userLocationChoices = [];
        for (let validLocation of validLocations) {
            if (formData[validLocation] === true) {
                userLocationChoices.push(validLocation);
            }
        }
        if (userLocationChoices.length !== locations.length) {
            console.log("No on locations--length!!! ðŸ˜«ðŸ˜«ðŸ˜«")
        }
        userLocationChoices.sort();
        console.log("userLocationChoices:", userLocationChoices);
        console.log("locations:", locations)
        let pointer1 = 0;
        let pointer2 = 0
        while (pointer1 < userLocationChoices.length) {            
            if (userLocationChoices[pointer1] !== locations[pointer2]) {
                console.log("No on locations--accuracy!!! ðŸ˜«ðŸ˜«ðŸ˜«");
                break;
            }
            else if (userLocationChoices[pointer1] === locations[pointer2]) {
                pointer1++;
                pointer2++;
            }
            // currently this runs for each correct response; is that what we want?
            console.log("YOU ARE CORRECT---locations!!! ðŸ˜€ðŸ˜€ðŸ˜€") 
        }
       
        console.log(formData)

        if (vertebrateGroup === taxClass && animalDiet === diet) {
            console.log("YOU ARE CORRECT---class and diet!!! ðŸ˜€ðŸ˜€ðŸ˜€");
        }
        else console.log("SORRY, NOT THIS TIME--class and diet!!! ðŸ˜«ðŸ˜«ðŸ˜«");     
    }
    // console.log(animal);
    // console.log(locations);
    // console.log(validLocations);
    
    const handleReset = () => {
        setFormData(initialState);
    }




    return (
        <div>
            <h1>Take the {commonName.toUpperCase()} challenge!</h1>
            <form onSubmit={handleSubmit}>
                
                {phylum === "Chordata" ?                
                <div>
                    <fieldset>
                        <legend>What taxonomic class does the {commonName.toLowerCase()} belong to?</legend>
                        <label>
                            <input 
                                type="radio"
                                name="vertebrateGroup"
                                value="Amphibia"
                                onChange={handleChange}
                                checked={vertebrateGroup === "Amphibia"}
                                >
                            </input>
                            Amphibians
                        </label>
                        <label>
                            <input 
                                type="radio"
                                name="vertebrateGroup"
                                value="Aves"
                                onChange={handleChange}
                                checked={vertebrateGroup === "Aves"}
                                >
                            </input>
                            Birds
                        </label>
                        <label>
                            <input 
                                type="radio"
                                name="vertebrateGroup"
                                value="Mammalia"
                                onChange={handleChange}
                                checked={vertebrateGroup === "Mammalia"}
                                >
                            </input>
                            Mammals
                        </label>
                        <label>
                            <input 
                                type="radio"
                                name="vertebrateGroup"
                                value="Reptilia"
                                onChange={handleChange}
                                checked={vertebrateGroup === "Reptilia"}
                                >
                            </input>
                            Reptiles
                        </label>
                        <label>
                            <input 
                                type="radio"
                                name="vertebrateGroup"
                                value="Fish"
                                onChange={handleChange}
                                checked={vertebrateGroup === "Fish"}
                                >
                            </input>
                            Fish
                        </label>
                    </fieldset>
                </div>

                : null }

                <fieldset>
                    <legend>Which term best describe's the {commonName.toLowerCase()}'s diet'?</legend>
                    <label>
                        <input 
                            type="radio"
                            name="animalDiet"
                            value="Carnivore"
                            onChange={handleChange}
                            >
                        </input>
                        Carnivore
                    </label>
                    <label>
                        <input 
                            type="radio"
                            name="animalDiet"
                            value="Herbivore"
                            onChange={handleChange}
                            >
                        </input>
                        Herbivore
                    </label>
                    <label>
                        <input 
                            type="radio"
                            name="animalDiet"
                            value="Omnivore"
                            onChange={handleChange}
                            >
                        </input>
                        Omnivore
                    </label>
                </fieldset>

                {validLocations.includes(locations[0]) ? 
                <div>
                    <fieldset>
                        <legend>In which location or locations can {commonName.toLowerCase()} be found?
                        </legend>
                        <label>
                            <input
                                type="checkbox"
                                id="Africa"
                                name="Africa"
                                checked={Africa}
                                onChange={handleChange}
                            ></input>
                            Africa
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Antarctica"
                                name="Antarctica"
                                checked={Antarctica}
                                onChange={handleChange}
                            ></input>
                            Antarctica
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Asia"
                                name="Asia"
                                checked={Asia}
                                onChange={handleChange}
                            ></input>
                            Asia
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Europe"
                                name="Europe"
                                checked={Europe}
                                onChange={handleChange}
                            ></input>
                            Europe
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="NorthAmerica"
                                name="NorthAmerica"
                                checked={NorthAmerica}
                                onChange={handleChange}
                            ></input>
                            North America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Eurasia"
                                name="Eurasia"
                                checked={Eurasia}
                                onChange={handleChange}
                            ></input>
                            Eurasia
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Ocean"
                                name="Ocean"
                                checked={Ocean}
                                onChange={handleChange}
                            ></input>
                            Ocean
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="CentralAmerica"
                                name="CentralAmerica"
                                checked={CentralAmerica}
                                onChange={handleChange}
                            ></input>
                            Central America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="SouthAmerica"
                                name="SouthAmerica"
                                checked={SouthAmerica}
                                onChange={handleChange}
                            ></input>
                            South America
                        </label>
                        <label>
                            <input
                                type="checkbox"
                                id="Oceania"
                                name="Oceania"
                                checked={Oceania}
                                onChange={handleChange}
                            ></input>
                            Oceania
                        </label>
                    </fieldset>
                </div> 
                : null}

                {/* add a bonus question of scientific name with text input */}
                
                <button type="submit">Submit</button>
                <button type="reset" onClick={handleReset}>Reset</button>
            </form>
        </div>
    )
}

/**
 * Functions to access data by category
 */
    /////////////////////////////////////////////////////////////////////
                // const [taxClasses, setTaxClasses] = useState([])
                // useEffect(() => {
                //     async function getClasses() {
                //         setTaxClasses(await AnimalsAPI.getTaxClasses());
                //     }
                //     getClasses()
                // }, []);
                // console.log(taxClasses)
    /////////////////////////////////////////////////////////////////////
    // Chondrichthyes: cartilagenous fish
    // Actinopterygii: ray-finned fish (bony)
    // Sarcopterygii: lobe-finned fish (bony)

    /////////////////////////////////////////////////////////////////////
                // const [allLocations, setAllLocations] = useState([]);
                // useEffect(() => {
                //     async function getAllLocations() {
                //         setAllLocations(await AnimalsAPI.getLocations());
                //     }
                //     getAllLocations();
                // }, []);

                // console.log(allLocations)
    /////////////////////////////////////////////////////////////////////

    /////////////////////////////////////////////////////////////////////
                // const [allHabitats, setAllHabitats] = useState([]);
                // useEffect(() => {
                //     async function getAllHabitats() {
                //         setAllHabitats(await AnimalsAPI.getHabitats());
                //     }
                //     getAllHabitats();
                // }, []);

                // console.log(allHabitats)
    /////////////////////////////////////////////////////////////////////