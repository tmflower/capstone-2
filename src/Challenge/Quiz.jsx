import { useEffect, useState } from 'react';
import { DietForm } from './DietForm';
import { LocationsForm } from './LocationsForm';
import { TaxClassForm } from './TaxClassForm';
import { ScientificNameForm } from './ScientificNameForm';
import { PreyForm } from './PreyForm';
import usersAPI from '../api/usersAPI';
import { useContext } from 'react';
import UserContext from '../userContext';
import { useNavigate, Link } from 'react-router-dom';
import validLocations from '../Animal/Animal';
import { Button, Modal, Box, Paper, IconButton, Typography, Stack } from '@mui/material';
import Close from "@mui/icons-material/Close";
import { theme } from "../theme/theme";
import gifs from '../assets/gifs';


/** Quiz component renders a set of questions about the selected animal;
 * User can check each answer individually;
 * Submit button only displays when 2 or more answers are correct;
 * User may go back to review facts on animal page, then return to Quiz;
 * On submit, user receives points and badge and is redirected to dashboard;
 * User will also see a congratulations messages with a celebration gif;
 */
const minNumQuestionsRequired = 2;

/** Quiz accepts props from Animal component; these provide the basis for the questions 
* Logged-in user is accessed with useContext;
* If user is not logged in, Quiz is not displayed; user will be prompted to sign up or log in
*/

export function Quiz({
                    taxClass,
                    setTaxClass, 
                    commonName, 
                    scientificName, 
                    locations, 
                    diet, 
                    phylum, 
                    prey, 
                    setAnimalSelected,
                    icon }) {

    const currentUser = useContext(UserContext);
    const username = currentUser.user.username;
    const userId = currentUser.user.id;
    const navigate = useNavigate();

    const message = {
        correct: "⭐️⭐️⭐️ You got it! ⭐️⭐️⭐️",
        incorrect: "😧😧😧 Sorry, try again. 😧😧😧"
    };

    const [points, setPoints] = useState(0);
    const [numQuestions, setNumQuestions] = useState(0);
    const [animalId, setAnimalId] = useState();
    const [animals, setAnimals] = useState([]);

    // Retrieve list of all animals in db;
    // We'll use this to create a badge for this user & animal;
    useEffect(() => {
        async function getAnimals() {
            const res = await usersAPI.getAllAnimals();
            setAnimals(res.animals);
        }
        getAnimals();
    }, [commonName]);

    // From the list of all animals, locate the selected animal by name;
    // Set animalId to be the id number in the animals table in db;
    // We'll use animalId to add a badge for this user & animal
    useEffect(() => {
        async function getAnimalId() {
            if(animals.length) {
                const animal = animals.filter(animal => commonName === animal.common_name);
                setAnimalId(animal[0].id);
            }
        } 
        getAnimalId();
    },[animals, commonName]);

    /** When user submits answers:
     * - Redirect user to dashboard
     * - Update user's points
     * - Update user's badges
     * - Provide congratulations message to user
     */
    async function handleSubmit() {
            await usersAPI.updatePoints({ username, points });
            await usersAPI.addBadge({ animalId, userId });
            handleOpen();
    }

    // Handle opening and closing of modal to with congratulations message when user earns badge;
    const [open, setOpen] = useState(false);
    const handleOpen = () => setOpen(true);
    const handleClose = () => {
    navigate("/dashboard", {replace: true});         
    }

    // This function allows user to return from quiz view to animal info view;  
    const handleClick = () => {
        setAnimalSelected(false);
    }
    
    // Selects a gif at random to display in modal when user earns badge;
    const randomNum = Math.floor(Math.random() * gifs.length);
    const gif = gifs[randomNum];

    return (
        <Box sx={{ mt: 5 }}>
            <Modal
                open={open}
                onClose={handleClose}
                aria-labelledby="modal-modal-title"
                aria-describedby="modal-modal-description"
            >
                <Box sx={{ display: 'flex', justifyContent: 'center', marginTop: '10rem' }}>
                <Paper 
                    className="popup" 
                    sx={{ padding: '2rem '}}>
                    <IconButton onClick={handleClose}>
                        <Close sx={{ color: theme.typography.primary }}/>
                    </IconButton>
                    <Stack>
                        <img 
                            id="modal-img"
                            src={gif} 
                            alt="celebration"/>
                        <Typography 
                            id="modal-modal-title" 
                            variant="h4" 
                            sx={{ fontFamily: theme.typography.primary, textAlign: 'center' }}>Congratulations, {username}!
                        </Typography>
                        <Typography 
                            id="modal-modal-description" 
                            variant="h5" 
                            sx={{ mt: 2, fontFamily: theme.typography.primary, textAlign: 'center' }}>You earned {points} points and the {commonName} badge!
                        </Typography>
                    </Stack>

                </Paper>
                </Box>
            </Modal>

            <Box sx={{ 
                    display: 'flex',
                    flexDirection: 'column', 
                    alignItems: 'center', 
                    textAlign: 'center'}}>
                <Typography 
                    variant="h3" 
                    sx={{ fontFamily: theme.typography.primary, m: 1 }}>Take the {commonName.toUpperCase()} challenge!
                </Typography>
                <img 
                    src={icon} 
                    alt={commonName} 
                    width="400px" 
                    className="animalIcon"/>
                <Typography 
                    variant="h5" 
                    sx={{ fontFamily: theme.typography.primary, m: 2 }}>Earn 10 points for each correct answer. Check each answer as you go. If you get it wrong, you can check the animal info and try again.
                </Typography>
                <img 
                    src={icon} 
                    alt={commonName} 
                    width="80px"/>
                <Typography 
                    variant="h5" 
                    sx={{ fontFamily: theme.typography.primary, m: 2 }}>When at least 2 answers are correct, the <span id="mock-btn">Submit Answers</span> button will appear at the bottom of the page.
                </Typography>
                <img 
                    src={icon} 
                    alt={commonName} 
                    width="80px"/>
                <Typography 
                    variant="h5" 
                    sx={{ fontFamily: theme.typography.primary, m: 2 }}>For more points and faster leveling up, answer any additional questions before you submit your answers.
                </Typography>
                <img 
                    src={icon} 
                    alt={commonName} 
                    width="80px"/>
                <Typography 
                    variant="h5" 
                    sx={{ fontFamily: theme.typography.primary, m: 2 }}>Visit your <Link to="/dashboard">dashboard</Link> to see your current level and how many points you need to level up.
                </Typography>
            </Box>

            <DietForm 
                commonName={commonName} 
                diet={diet}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                >
            </DietForm>

            <LocationsForm 
                commonName={commonName} 
                locations={locations}
                message={message}
                points={points}
                setPoints={setPoints}
                numQuestions={numQuestions}
                setNumQuestions={setNumQuestions}
                validLocations={validLocations}
                >
            </LocationsForm>

            {phylum === "Chordata" ?  
                <TaxClassForm 
                    commonName={commonName} 
                    taxClass={taxClass} 
                    setTaxClass={setTaxClass} 
                    phylum={phylum}
                    message={message}
                    points={points}
                    setPoints={setPoints}
                    numQuestions={numQuestions}
                    setNumQuestions={setNumQuestions}
                    >
                </TaxClassForm> 
            : null} 

            { prey ? 
                <PreyForm 
                    commonName={commonName} 
                    prey={prey}
                    message={message}
                    points={points}
                    setPoints={setPoints}
                    numQuestions={numQuestions}
                    setNumQuestions={setNumQuestions}
                    >
                </PreyForm> 
            : null }

            {scientificName ?
                <ScientificNameForm 
                    commonName={commonName} 
                    scientificName={scientificName}
                    message={message}
                    points={points}
                    setPoints={setPoints}
                    numQuestions={numQuestions}
                    setNumQuestions={setNumQuestions}
                    >
                </ScientificNameForm>  
            : null }   

            <Button 
                type="button" 
                onClick={handleClick} 
                id="alt-button">Return to animal info
            </Button>
            { numQuestions >= minNumQuestionsRequired ?           
                <Button 
                    type="submit" 
                    onClick={handleSubmit} 
                    id="alt-button">Submit Your Answers
                </Button> 
            : null }
        </Box>
    )
}
