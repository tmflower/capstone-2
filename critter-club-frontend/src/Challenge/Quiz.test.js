import { render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from 'react-router-dom';
import { Quiz } from './Quiz';
import UserContext from '../userContext';

// smoke test
test("renders without crashing", () => {
    const currentUser = {user: { username: "Tiny", id: 1, password: 'abc123$%^', points: 40, userBadges: ['Koala', 'Jellyfish', 'Painted Turtle'], parentId: 1 }}
    render(
    <MemoryRouter>
      <UserContext.Provider value={currentUser}>
        <Quiz 
          commonName="Hippopotamus"
          locations="some location"/>
      </UserContext.Provider>      
    </MemoryRouter>);
  });

// snapshot test
test("matches snapshot", () => {
  const currentUser = {user: { username: "Tiny", id: 1, password: 'abc123$%^', points: 40, userBadges: ['Koala', 'Jellyfish', 'Painted Turtle'], parentId: 1 }}
    const { asFragment } = render(
      <MemoryRouter>
        <UserContext.Provider value={currentUser}>
          <Quiz 
            commonName="Hippopotamus"
            locations="some location"/>
        </UserContext.Provider>        
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
});

// tests for user interactions

// test("changes from quiz view to animal info view when user clicks back button", async () => {
//   const currentUser = {user: { username: "Tiny", id: 1, password: 'abc123$%^', points: 40, userBadges: ['Koala', 'Jellyfish', 'Painted Turtle'], parentId: 1 }}
// const user = userEvent.setup();  
//   render(  
//     <MemoryRouter>
//       <UserContext.Provider value={currentUser}>
//         <Quiz 
//         commonName="Hippopotamus"
//         locations="some location"
//         />
//       </UserContext.Provider>      
//     </MemoryRouter>);
    
//   expect(screen.getByText('Take the HIPPOPOTAMUS challenge!')).toBeInTheDocument();
//   expect(screen.queryByText('Collecting your animal information!')).not.toBeInTheDocument();

//   await user.click(screen.getByText("Return to animal info"));

//   expect(screen.queryByText('Take the HIPPOPOTAMUS challenge!')).not.toBeInTheDocument();
//   expect(screen.getByText('Collecting your animal information!')).toBeInTheDocument();
// });