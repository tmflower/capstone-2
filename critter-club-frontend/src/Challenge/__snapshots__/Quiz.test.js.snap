// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-y1gt6f"
  >
    <div
      class="MuiBox-root css-gdx2i7"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-plqo20-MuiTypography-root"
      >
        Take the HIPPOPOTAMUS challenge!
      </h3>
      <img
        alt="Hippopotamus"
        class="animalIcon"
        width="400px"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-gk9nbu-MuiTypography-root"
      >
        Earn 10 points for each correct answer. Check each answer as you go. If you get it wrong, you can check the animal info and try again.
      </h5>
      <img
        alt="Hippopotamus"
        width="80px"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-gk9nbu-MuiTypography-root"
      >
        When at least 2 answers are correct, the 
        <span
          id="mock-btn"
        >
          Submit Answers
        </span>
         button will appear at the bottom of the page.
      </h5>
      <img
        alt="Hippopotamus"
        width="80px"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-gk9nbu-MuiTypography-root"
      >
        For more points and faster leveling up, answer any additional questions before you submit your answers.
      </h5>
      <img
        alt="Hippopotamus"
        width="80px"
      />
      <h5
        class="MuiTypography-root MuiTypography-h5 css-gk9nbu-MuiTypography-root"
      >
        Visit your 
        <a
          href="/dashboard"
        >
          dashboard
        </a>
         to see your current level and how many points you need to level up.
      </h5>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-2e4xct-MuiPaper-root"
    >
      <form>
        <fieldset>
          <div
            class="MuiFormGroup-root css-dmmspl-MuiFormGroup-root"
            role="radiogroup"
          >
            <label
              class="MuiFormLabel-root MuiFormLabel-colorPrimary css-tshmje-MuiFormLabel-root"
            >
              <legend>
                Which term best describes the hippopotamus's diet?
              </legend>
            </label>
            <label
              for="Carnivore"
            >
              <span
                class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  id="Carnivore"
                  name="animalDiet"
                  type="radio"
                  value="Carnivore"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Carnivore
            </label>
            <label
              for="Herbivore"
            >
              <span
                class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  id="Herbivore"
                  name="animalDiet"
                  type="radio"
                  value="Herbivore"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Herbivore
            </label>
            <label
              for="Omnivore"
            >
              <span
                class="MuiButtonBase-root MuiRadio-root MuiRadio-colorPrimary PrivateSwitchBase-root MuiRadio-root MuiRadio-colorPrimary MuiRadio-root MuiRadio-colorPrimary css-vqmohf-MuiButtonBase-root-MuiRadio-root"
              >
                <input
                  class="PrivateSwitchBase-input css-1m9pwf3"
                  id="Omnivore"
                  name="animalDiet"
                  type="radio"
                  value="Omnivore"
                />
                <span
                  class="css-hyxlzm"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hbvpl3-MuiSvgIcon-root"
                    data-testid="RadioButtonUncheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                    />
                  </svg>
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1hhw7if-MuiSvgIcon-root"
                    data-testid="RadioButtonCheckedIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </span>
              Omnivore
            </label>
            <div>
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="submit"
              >
                Check answer
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
            <h5
              class="MuiTypography-root MuiTypography-h5 css-18czui-MuiTypography-root"
              id="quiz-feedback"
            />
          </div>
        </fieldset>
      </form>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 css-2e4xct-MuiPaper-root"
    >
      <form />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-1e6y48t-MuiButtonBase-root-MuiButton-root"
      id="alt-button"
      tabindex="0"
      type="button"
    >
      Return to animal info
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
    <a
      href="/dashboard"
    />
  </div>
</DocumentFragment>
`;
